Unit Tests for mtbl_merge

kadealicious opened the pull request:
- Add a unit test for `mtbl_merge`.  This adds 3 `.data` files and a `.so` file containing the merge function.
- Increase coverage for threadpool, writer, and sorter.
- Formatting.

kadealicious committed:
Add a threaded pass to sorted merge test

kadealicious committed:
Explicitly specify compression level in compression test

kadealicious committed:
Add unit tests for mtbl_merge

kadealicious committed:
Remove unnecessary block-size from sorted merge test

kadealicious committed:
Explicitly specify sorter max memory in sorted merge test

reedjc commented:
Instead of updating your branch I created a new branch unit-tests-reed-202501
Please see the commit there.
Here are a few comments:

The open source mtbl code is used on various platforms so we cannot ship the ELF x86-64 64-bit LSB shared object. So we build it instead. Since we don't know name of the built file (like MacOS vs Linux) also generate the script that sets the DSO name.

Instead of assuming mtbl_info lines in some order, we can check for specific lines.

I am unsure how the commas for the 1,024 and 65,536 worked for you. I looked at the mtbl_info printf code. I removed the commas.

Also use top_builddir instead of top_srcdir for generated files.

There was a -Wmissing-prototypes warning so I added  the prototype.

See the commit message and the commit for more details.

I didn't test this on non-Linux yet nor did I test in our buildfarm yet.

kadealicious commented:
> Instead of updating your branch I created a new branch unit-tests-reed-202501 Please see the commit there. Here are a few comments:
> 
> The open source mtbl code is used on various platforms so we cannot ship the ELF x86-64 64-bit LSB shared object. So we build it instead. Since we don't know name of the built file (like MacOS vs Linux) also generate the script that sets the DSO name.
> 
> Instead of assuming mtbl_info lines in some order, we can check for specific lines.
> 
> I am unsure how the commas for the 1,024 and 65,536 worked for you. I looked at the mtbl_info printf code. I removed the commas.
> 
> Also use top_builddir instead of top_srcdir for generated files.
> 
> There was a -Wmissing-prototypes warning so I added the prototype.
> 
> See the commit message and the commit for more details.
> 
> I didn't test this on non-Linux yet nor did I test in our buildfarm yet.

Cherry-picked this commit here.  `mtbl_info` was outputting values with commas on my machine, so also `export LANG=C` within the merge test.
